tools/topology/topology2/development/tplg-targets.cmake

```diff
@@ -96,7 +96,7 @@ DMIC0_PCM_CHANNELS=4"
 HDA_CONFIG=efx,NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,EFX_FIR_PARAMS=passthrough,\
 EFX_IIR_PARAMS=passthrough,EFX_DRC_COMPONENT=multiband,EFX_MBDRC_PARAMS=passthrough,\
 DMIC0_ENHANCED_CAPTURE=true,EFX_DMIC0_TDFB_PARAMS=line4_generic_pm10deg,\
-EFX_DMIC0_DRC_PARAMS=dmic_default,DMIC0_PCM_CHANNELS=2"
+EFX_DMIC0_DRC_PARAMS=dmic_default,DMIC0_PCM_CHANNELS=2,DMIC0_SAME_CHANNELS=false"
```

Comment:
Would not be more user friendly if you would:
Set the default
```
DMIC0_PCM_CHANNELS		0
```
then have the magic:
```
IncludeByKey.DMIC0_PCM_CHANNELS {
	"0" {
		Define {
			DMIC0_PCM_CHANNELS NUM_DMICS
		}
	}
}
```
So, if only the `NUM_DMICS` is set then the `DMIC0_PCM_CHANNELS` would default to the same value, however if both is set then they will keep their set values, which would allow:
stereo DMIC with stereo PCM: `NUM_DMICS=2`
4 DMIC with 4ch PCM: `NUM_DMICS=4`
4 DMIC with 2ch PCM: `NUM_DMICS=4,DMIC0_PCM_CHANNELS=2`

Reply:
True, it saves one macro to worry about. I'll try if it can be done (setting macro from macro -- or what are these called in this context ...).

Reply:
Setting it with another IncludeByKey works, testing it.